### PR TITLE
Escape module names when fetching details json

### PR DESCRIPTION
**Pull request by asdil12**
This prevents issues with modules with a `#1` suffix

Ticket: https://progress.opensuse.org/issues/95033

### CODE REVIEW

**Review by perlpunk — COMMENTED** [review]
Can you add a test?

**Comment by okurz** [comment]
> Can you add a test?

maybe just rename existing soft fail reference to have a "#1" included in the module name

**Review by okurz — COMMENTED** [review]
For a test please try

```diff

```--- a/tests/tags_labels/:tests:1316928
+++ b/tests/tags_labels/:tests:1316928
@@ -1,7 +1,7 @@
 <tr>
     <td class="component">
         <div>
-            <a href="/tests/1316928/modules/shutdown/steps/1/src">shutdown</a>
+            <a href="/tests/1316928/modules/shutdown%231/steps/1/src">shutdown#1</a>
         </div>
         <div class="flags">
         </div>
@@ -12,7 +12,7 @@
     <td class="links">
         <div class="links_a">
             <div class="fa fa-caret-up"></div>
-            <a class="no_hover" data-url="/tests/1316928/modules/shutdown/steps/9" href="#step/shutdown/9"
+            <a class="no_hover" data-url="/tests/1316928/modules/shutdown%231/steps/9" href="#step/shutdown/9"
                title="Soft Failed">
                 <span class="resborder resborder_hash(0x9c726c8)">Soft Failed</span>
             </a></div>

File: tests/test_openqa_review.py [hunk 1]
```diff
@@ -478,6 +478,7 @@ def test_bugrefs_with_report_links():
 def test_bugrefs_including_softfails():
     args = bugrefs_test_args_factory()
     args.include_softfails = True
+    args.verbose = 5
```
Comment (okurz) [inline tests/test_openqa_review.py:481]: please exclude that from your commit. If increasing the verbosity level makes sense then please in a separate commit + separate PR + not only for this subtest

**Review by okurz — CHANGES_REQUESTED** [review]

File: tests/test_openqa_review.py (hunk 1 above)
Comment (asdil12) [reply tests/test_openqa_review.py:481]: when I add it globally for the factory, it breaks the testsuite.
I thought it makes sense here as it makes this very issue visible.

Comment (asdil12) [reply tests/test_openqa_review.py:481]: but I can leave it out for now and we can work in another PR on increasing the loglevel for the testsuite.

**Review by Martchus — APPROVED** [review]

**Pull request merged**